core/src/main/java/org/elasticsearch/common/settings/AbstractScopedSettings.java

```diff
@@ -410,4 +414,18 @@ private static final boolean applyDeletes(Set<String> deletes, Settings.Builder
         return changed;
     }
 
+    private static Setting<?> findOverlappingSettng(Setting<?> newSetting, Map<String, Setting<?>> complexMatchers) {
```

Comment:
Nit: There is a typo in the method name. The probably intended name is `findOverlappingSetting` (the "i" is missing)